### ISSUE TITLE
Update CHANGES

### ISSUE THREAD

**Issue by olexandr-konovalov** [gap-packages/FactInt#11]
Reminder: update CHANGES by documenting what has been done for 1.6.1 when there will be time to make 1.6.2.

**Comment by olexandr-konovalov** [comment]
@Stefan-Kohl I am going to publish 1.6.2 now - is that OK?
```
Changes between FactInt 1.6.2 and FactInt 1.6.1 (February 17, 2018)

 - Rewritten `FactorsTDNC` to avoid recursion.
 - Some optimisations of loading data tables.

Changes between FactInt 1.6.0 and FactInt 1.6.1 (January 17, 2018)

 - fixed a bug in `FactorsECM` that may result in passing a non-integral
   fraction to `RootInt`. Now we ensure that the 1st argument of `RootInt`
   is converted to an integer first.

```

**Comment by Stefan-Kohl** [comment]
@alex-konovalov In the description of the bug, I'd suggest to explain why the bug didn't have actual consequences -- i.e. why it has never led to error messages, let alone wrong results. Otherwise the message sounds unnecessarily alarming. -- Apart from that, please go ahead and publish 1.6.2!

**Comment by olexandr-konovalov** [comment]
@Stefan-Kohl which text you prefer then? Is it actually guaranteed that it "never led to error messages, let alone wrong results" or just no such ones are known. I am looking at https://github.com/gap-system/gap/pull/2061 and it's not so all clear to me...

**Comment by Stefan-Kohl** [comment]
@alex-konovalov So far, RootInt didn't error out if the argument was a fraction -- so FactorsECM also didn't.  -- I'd write something like:


    Changes between FactInt 1.6.0 and FactInt 1.6.1 (January 17, 2018)

     - The function `FactorsECM` has been updated to reflect the change that
       `RootInt` no longer accepts non-integral arguments. Now we ensure that
       the 1st argument of `RootInt` is converted to an integer first.

**Comment by olexandr-konovalov** [comment]
The [documentation of RootInt in GAP 4.8.10](http://www.gap-system.org/Manuals/doc/ref/chap14.html#X83D9B5C87EEA2A77) says that the 1st argument is an integer, so it does look like a bug that it was called with the wrong argument from FactInt (and this is the label for PR #6 which fixes it).

**Comment by Stefan-Kohl** [comment]
@alex-konovalov My concern is to reserve the term 'bug' as a signal word for the user to denote something which may have negatively affected their work. If you use that term in a too technical sense and possibly frequently in contexts where users were not affected, it no longer serves as such signal word. -- Of course one could then introduce terms like 'dangerous bug' or whatsoever, and start an inflation of words -- but is this what we want?

**Comment by olexandr-konovalov** [comment]
I am not familiar with the code. Could you please explain what is the purpose of lines 
https://github.com/gap-packages/FactInt/blob/c3c08793369cd596b0ac0988e0942167411ef8cb/lib/ecm.gi#L316-L318

In https://github.com/gap-system/gap/pull/2061 you said that 
> In terms of FactInt, 1 was just as sensible as 0 -- and even 2 or 3 would hardly do more harm than causing a little slow-down on average ... . On the other hand, something like e.g. 100 would cause pretty poor performance in many cases. 

Does that mean that it had only impact on the performance, but not on the correctness?

**Comment by Stefan-Kohl** [comment]
@alex-konovalov 'Curves' is merely the number of curves to be tried, and the formula serves the purpose to give a value which is reasonable on average. Given that GAP uses now GMP, which makes ECM quite a bit faster, but does not proportionally affect the performance of the MPQS, it may well make sense to choose a higher value now than it was in the times when I have written the code. -- But properly calibrating this takes time. In any case, a bad value will only have an impact on performance.

**Comment by olexandr-konovalov** [comment]
@Stefan-Kohl ok, thanks for further details. Released 1.6.2 with your text from above in the changelog.
